src/uct/ib/dc/dc_mlx5.c

```diff
@@ -808,45 +809,79 @@ void uct_dc_mlx5_destroy_dct(uct_dc_mlx5_iface_t *iface)
 }
 #endif
 
-void uct_dc_mlx5_iface_dcis_destroy(uct_dc_mlx5_iface_t *iface, int max)
+static void uct_dc_mlx5_iface_dci_pool_destroy(uct_dc_mlx5_dci_pool_t *dci_pool)
+{
+    ucs_arbiter_cleanup(&dci_pool->arbiter);
+    ucs_free(dci_pool->stack);
+}
+
+static void uct_dc_mlx5_iface_dcis_destroy(uct_dc_mlx5_iface_t *iface,
+                                           uint8_t num_dci_pools,
+                                           uint8_t num_dcis)
 {
     uct_ib_mlx5_md_t *md = ucs_derived_of(iface->super.super.super.super.md,
                                           uct_ib_mlx5_md_t);
-    int i;
+    uint8_t pool_index, dci_index;
+
+    ucs_assert(num_dci_pools <= iface->tx.num_dci_pools);
+    ucs_assert(num_dcis <= uct_dc_mlx5_iface_total_ndci(iface));
 
-    for (i = 0; i < max; i++) {
-        uct_rc_txqp_cleanup(&iface->super.super, &iface->tx.dcis[i].txqp);
-        uct_ib_mlx5_destroy_qp(md, &iface->tx.dcis[i].txwq.super);
+    for (dci_index = 0; dci_index < num_dcis; dci_index++) {
+        uct_rc_txqp_cleanup(&iface->super.super,
+                            &iface->tx.dcis[dci_index].txqp);
+        uct_ib_mlx5_destroy_qp(md, &iface->tx.dcis[dci_index].txwq.super);
+    }
+
+    for (pool_index = 0; pool_index < num_dci_pools; pool_index++) {
+        uct_dc_mlx5_iface_dci_pool_destroy(&iface->tx.dci_pool[pool_index]);
     }
 
     ucs_free(iface->tx.dcis);
 }
 
-ucs_status_t
-uct_dc_mlx5_iface_create_dcis(uct_dc_mlx5_iface_t *iface,
+static ucs_status_t
+uct_dc_mlx5_iface_dcis_create(uct_dc_mlx5_iface_t *iface,
                               const uct_dc_mlx5_iface_config_t *config)
 {
-    uint8_t num_paths = iface->super.super.super.num_paths;
+    const uint8_t num_paths = iface->super.super.super.num_paths;
+    uint8_t dci_index       = 0;
     uct_dc_mlx5_dci_pool_t *dci_pool;
-    int i, pool_index, dci_index;
+    uint8_t pool_index, i;
     ucs_status_t status;
 
-    dci_index = 0;
+    /* allocate dcis arrays and stack */
```

Comment:
imo, no need for this comment (plz start with uppercase if decide to leave it).

Reply:
removed